lib/daru/core/query.rb

```diff
@@ -39,7 +39,7 @@ def inspect
 
       class << self
         def apply_scalar_operator operator, data, other
-          BoolArray.new data.map { |d| !!d.send(operator, other) }
+          BoolArray.new data.map { |d| !!d.send(operator, other) unless d.nil? }
```

Comment:
@v0dro `Float::NaN` I think are only used in numerical domain. I don't think they ever occur with strings or any other object except numbers.